qiskit/pulse/macros.py

```diff
@@ -92,6 +153,68 @@ def measure(
     return schedule
 
 
+def _measure_v2(
+    qubits: List[int],
+    target: Target,
+    meas_map: Union[List[List[int]], Dict[int, List[int]]],
+    qubit_mem_slots: Dict[int, int],
+    measure_name: str = "measure",
+) -> Schedule:
+    """Return a schedule which measures the requested qubits according to the given
+    target and measure map, or by using the defaults provided by the backendV2.
+
+    Args:
+        qubits: List of qubits to be measured.
+        target: The :class:`~.Target` representing the target backend.
+        meas_map: List of sets of qubits that must be measured together.
+        qubit_mem_slots: Mapping of measured qubit index to classical bit index.
+        measure_name: Name of the measurement schedule.
+
+    Returns:
+        A measurement schedule corresponding to the inputs provided.
+    """
+    schedule = Schedule(name=f"Default measurement schedule for qubits {qubits}")
+
+    if isinstance(meas_map, list):
+        meas_map = utils.format_meas_map(meas_map)
+    meas_group = set()
+    for qubit in qubits:
+        meas_group |= set(meas_map[qubit])
+    meas_group = sorted(list(meas_group))
+
+    meas_group_set = set(range(max(meas_group) + 1))
+    unassigned_qubit_indices = list(meas_group_set - qubit_mem_slots.keys())
```

Comment:
```suggestion
    unassigned_qubit_indices = list(set(meas_group) - qubit_mem_slots.keys())
```
I think this should work because qubit outside this group doesn't appear.